src/languages/scala.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  chainedMatcher,
+} from '../util/nodeMatchers';
+import { NodeMatcherAlternative, ScopeType } from '../typings/Types';
+
+// Generated by the following command:
+// > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-scala/master/src/node-types.json | jq '[.[] | select(.type == "_definition" or .type == "_expression" or .type == "_pattern") | .subtypes[].type]'
+const STATEMENT_TYPES = [
+  // FIXME: is this right? What is a "statement" in a language where most traditional statements are expressions
```

Comment:
As you suggest here, this list is very likely not what you want 😄. Notice for example that it has "identifier", so it would rarely expand very far. 

If you do want to support "statement" for Scala, I'd maybe start with just declaration types?

Fwiw I didn't define any statement scope type when I did clojure

Reply:
Yeah I'm going to swing back around to this one once I'm more familar with what the master branch generates

Reply:
I have decided to just not implement it for now. The more I use the scala bindings the more I can work out where they are weak and adjust accordingly